### PR TITLE
[Doc] Improve metrics aggregation Train user guide

### LINKED ISSUE #31434: [Train] Update documentation to show how to use `torchmetrics` for metric aggregation
### Description

For some applications of Ray Train, it may be desirable to report metrics from all workers and/or report aggregations, such as mean and std. While Train currently reports metrics only from rank 0 worker, Torch users can use `torchmetrics` to aggregate metrics from other workers and report them on rank 0. We should update the documentation to show how that can be done.

### Use case

_No response_

### PR DESCRIPTION
**Pull request by Yard1**
Signed-off-by: Antoni Baum <antoni.baum@protonmail.com>

## Why are these changes needed?

Improve the section concerning metrics aggregation from workers in the Train DL User Guide.

## Related issue number

Closes https://github.com/ray-project/ray/issues/31434

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/train/dl_guide.rst [hunk 1]
```diff
@@ -905,35 +906,103 @@ A simple example for creating a callback that will print out results:
...
-    import torch
-    import torchmetrics
+    .. code-block:: python
 
-    def train_func(config):
-        preds = torch.randn(10, 5).softmax(dim=-1)
-        target = torch.randint(5, (10,))
-        accuracy = torchmetrics.functional.accuracy(preds, target).item()
-        session.report({"accuracy": accuracy})
+        import ray.train.torch
```
Comment (richardliaw) [inline doc/source/train/dl_guide.rst:927]: Can you make this a tested python file?

Also, can you explicitly add a comment that says `First, pip install torchmetrics` and `this file is tested with torchmetrics=1.1.1`

File: doc/source/train/dl_guide.rst [hunk 2]
```diff
@@ -905,35 +906,103 @@ A simple example for creating a callback that will print out results:
     # {'epoch': 1, '_timestamp': 1656349412, '_time_this_iter_s': 0.0013833045959472656, '_training_iteration': 2, 'time_this_iter_s': 0.016670703887939453, 'done': False, 'timesteps_total': None, 'episodes_total': None, 'training_iteration': 2, 'trial_id': '0f1d0_00000', 'experiment_id': '494a1d050b4a4d11aeabd87ba475fcd3', 'date': '2022-06-27_17-03-32', 'timestamp': 1656349412, 'time_total_s': 3.4501540660858154, 'pid': 23018, 'hostname': 'ip-172-31-43-110', 'node_ip': '172.31.43.110', 'config': {}, 'time_since_restore': 3.4501540660858154, 'timesteps_since_restore': 0, 'iterations_since_restore': 2, 'warmup_time': 0.003779172897338867, 'experiment_tag': '0'}
 
 
-Example: PyTorch Distributed metrics
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. _train-aggregating-results:
 
-In real applications, you may want to calculate optimization metrics besides
-accuracy and loss: recall, precision, Fbeta, etc.
+Obtaining and aggregating results from multiple workers
```
Comment (richardliaw) [inline doc/source/train/dl_guide.rst:911]: Can we do like, `How to obtain and aggregate results from different workers?`

**Review by richardliaw — CHANGES_REQUESTED** [review]
nice!

File: doc/source/train/doc_code/torchmetrics_example.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
+# flake8: noqa
+# isort: skip_file
+
+# __start__
+
+# First, pip install torchmetrics
+# This code is tested with torchmetrics==1.1.1
```
Comment (richardliaw) [inline doc/source/train/doc_code/torchmetrics_example.py:7]: wait is it actually tested with 1.1.1?

**Review by richardliaw — APPROVED** [review]

File: doc/source/train/doc_code/torchmetrics_example.py (hunk 3 above)
Comment (richardliaw) [reply doc/source/train/doc_code/torchmetrics_example.py:7]: most recent version is 0.11.0
